### ISSUE TITLE
no_adjacent_strings_in_list should allow adjacent strings in single-element lists

### ISSUE THREAD

**Issue by srawlins** [dart-lang/linter#1272]
E.g.

```dart
var errors = [
    'This is a lengthy error message which '
    'takes up multiple lines, and I prefer to '
    'instantiate it first in a list.'
];

**Comment by srawlins** [comment]
This is seen a lot in Angular's `styleUrls`, which is almost always given a single string, and that single string is sometimes a long path.

```dart
@Component(
    styleUrls: [
    'package:foo.bar.baz.so.many.packages.deep'
        '/something.scss.css'
  ],
)
```

**Comment by srawlins** [comment]
CC @matanlurey @TedSander, do y'all have any preference or guidance on long Strings in Angular annotations? Does this come up a lot, where these String literals extend past 80 chars?

**Comment by bwilkerson** [comment]
Sorry I didn't notice this before now, but the purpose of this lint is to catch possible missing commas. If we find a list like `['a' 'b']` we can't tell whether the list is intended to have one element or two. I think this issue is asking us to infer an intent that we cannot possibly infer. I think changing the rule is counter to the original reason for the rule.

**Comment by srawlins** [comment]
Mm that's true. OK, I'll close. I think `+` is a good fix cases like this.
